consensus/state.go

```diff
@@ -1585,7 +1585,10 @@ func (cs *State) finalizeCommit(height int64) {
 		// but may differ from the LastCommit included in the next block
 		precommits := cs.Votes.Precommits(cs.CommitRound)
 		seenCommit := precommits.MakeCommit()
-		cs.blockStore.SaveBlock(block, blockParts, seenCommit)
+		err := cs.blockStore.SaveBlock(context.TODO(), block, blockParts, seenCommit)
```

Comment:
previously, I was using the `consensus.State.proposalCtx` here, but it was already canceled or set to nil before finalizing. This nil/cancelled proposal context was being passed to `PutBlock` which was causing weird things to happen with IPFS. It will just hang indefinitely, spinning up tons goroutines. Still need to figure out precisely what causes it and write a test/check, but yeah.

Reply:
I think in case we move block saving from `defaultSetProposal` to `finalizeCommit` - context logic should be reworked

Reply:
I think it is ok to use TODO there for now, as it just saves things locally. Also, we don't need proposal context there as they are different operations.